src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -260,7 +260,7 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
         public override void VisitDeclarationExpression(DeclarationExpressionSyntax node)
         {
             var argumentSyntax = (ArgumentSyntax)node?.Parent;
-            var argumentListSyntax = (BaseArgumentListSyntax)argumentSyntax?.Parent;
+            var argumentListSyntax = argumentSyntax.FirstAncestorOrSelf<BaseArgumentListSyntax>();
```

Comment:
Yes, it was wrong now that declaration expressions can appear in new and novel contexts. However, perhaps it would be better to replace the (incorrect) cast with an `as` operator.